variod.c

```diff
@@ -289,112 +350,107 @@ int main(int argc, char *argv[])
 		// since we might have started after XCSoar was running for a while
 		request_current_settings = true;
 
-		//Receive a message from sensord and forward to XCsoar
-		while ((read_size = recv(connfd, client_message, 2000, 0 )) > 0 ) {
+		// receive a message from sensord and forward to XCsoar
+		while ((read_size = recv(connfd, client_message, sizeof(client_message), 0 )) > 0 ) {
+			// client_message may hold several sentences separated by a '\0'.
```

Comment:
My mute problem starts when getting a read_size=0 here what closes, the socket an unmutes vario in the while (1) loop

Reply:
Are you sure you're not getting `read_size==-1` ?
I've never seen read_size==0 . The NONBLOCKING flag is removed. Line 259.

Reply:
Completely sure, look at the logs I've posted.
You can see 0 and POV sentences being received partially and completely. No -1 (error) anywhere.
read_size=0 is logged just before close(connfd) out of the read loop, POV sentences and bytes read just after the while inside the read loop.
0 means the peer closed the connection cleanly from what I could research.
I'm trying to contact Andreas, who helped with the porting of Variod to see if he can help with sensord.
The problem lied there all along from my point of view. If you connect XCSoar to sensord this behaviour also occurs ruling out variod as the problem.

Reply:
Tried making this a blocking socket too just to test. Did not work.
Making the socket blocking adds sleep time to the command scanning routine.
You may start loosing commands that way. Perhaps it is not noticeable, perhaps it is.
Don´t know. Till we don't get rid of those zeroes all other stuff is second priority.

Reply:
I have no H/W with sensors on. So I can't run sensord.
Have you tried with netcat ? Or with sensord_mock ? Would be interesting to see if the connection closes there, too. Here it runs for hours w/o any issues. I replay 50000 NMEA sentences multiple times on the OV machine as well as on debian 10.

Reply:
I have actual sensors in my OV. I don't run the software on a PC. All my tests are with the actual hardware.
Tried to debug sensord, but the mere fact of adding debug_print() inside the main loop made it work as it should. So I can not make it fail while looking. That's why I'm trying to reach the original writers to ask how they did to time everything and why it changed when we changed the Operating System. Don't forget this worked flawlessly in Armstrong as it was.
Both state machines running in sensord are there to time sensor measure and read (need delay between these actions). Both state machines are made of switch/cases. If that timing changes don't know what the result might be. Let's hope Andreas can help us understand how to test and debug sensord.